pegasus/sites.v3/code.org/public/ops-dashboard/index.haml

```diff
@@ -0,0 +1,376 @@
+    $routeProvider.when('/workshops',                                 {templateUrl: '/ops-dashboard/workshops',       controller: 'WorkshopsController'});
+    $routeProvider.when('/workshops/new',                             {templateUrl: '/ops-dashboard/workshop-edit',   controller: 'WorkshopController'});
+    $routeProvider.when('/workshops/:id/edit',                        {templateUrl: '/ops-dashboard/workshop-edit',   controller: 'WorkshopController'});
+    $routeProvider.when('/workshops/:workshopid/attendance',          {templateUrl: '/ops-dashboard/attendance',      controller: 'AttendanceController'});
+    $routeProvider.when('/cohorts',                                   {templateUrl: '/ops-dashboard/cohorts',         controller: 'CohortsController'});
+    $routeProvider.when('/cohorts/new',                               {templateUrl: '/ops-dashboard/cohort-edit',     controller: 'CohortController'});
+    $routeProvider.when('/cohorts/:id/edit',                          {templateUrl: '/ops-dashboard/cohort-edit',     controller: 'CohortController'});
+    $routeProvider.otherwise({redirectTo: '/'});
+  }]).config(['$httpProvider', function($httpProvider) {
+    // X-Requested-With header required for CSRF requests protected by Rack::Protection::JsonCsrf included by Sinatra.
+    // Angular originally set this, but removed it in a breaking change in v1.4 because it is "rarely used in practice":
+    // https://github.com/angular/angular.js/commit/3a75b1124d062f64093a90b26630938558909e8d
+    $httpProvider.defaults.headers.common["X-Requested-With"] = 'XMLHttpRequest';
+  }]);
+
+  // SERVICES
+  var services = angular.module('opsApp.services', []).
+      value('version', '0.1');
+
+  services.factory('districtsService', ['$resource',
+    function($resource) {
+      var District = $resource('/dashboardapi/districts/:id', {}, {
+        // default methods: see https://code.angularjs.org/1.2.21/docs/api/ngResource/service/$resource
+        //  'get':    {method: 'GET'},
+        //  'save':   {method: 'POST'},
+        //  'query':  {method: 'GET', isArray:true},
+        //  'remove': {method: 'DELETE'},
+        //  'delete': {method: 'DELETE'} // don't use this because it doesn't work in IE9
+        update: {method:'PUT', url: '/dashboardapi/districts/:id'}
+      });
+
+      return District;
+    }]);
+  services.factory('programsService', [
+    function() {
+      return [{name:'MSS'}, {name:'MSM'}, {name:'ECS'}, {name:'CSP'}];
+  }]);
+
+  services.factory('workshopsService', ['$resource',
+    function($resource){
+      var Workshop = $resource('/dashboardapi/workshops/:id', {}, {
+        // default methods: see https://code.angularjs.org/1.2.21/docs/api/ngResource/service/$resource
+        //  'get':    {method:'GET'},
+        //  'save':   {method:'POST'},
+        //  'query':  {method:'GET', isArray:true},
+        //  'remove': {method:'DELETE'},
+        //  'delete': {method:'DELETE'}
```

Comment:
I've been putting these in angular services as documentation for what the built in methods in $resource are